scripts/macomber_to_csv.py

```diff
@@ -0,0 +1,299 @@
+#!/usr/bin/env python
+
+'''
+Convert structured text file based on Macomber's handlist
+and CSV of incipits into a set of CSV files for manuscripts,
+canonical stories, and story instances.
+
+Input files are included in the data directory
+
+Usage::
+
+    python ./scripts/macomber_to_csv.py -f data/macomber-miracles.txt -i data/incipits.csv
+
+'''
+
+
+import argparse
+import csv
+import json
+import os
+import re
+from collections import OrderedDict, defaultdict
+
+
+SCRIPT_DIR = os.path.dirname(__file__)
+
+
+class MacomberToCsv:
+
+    # regex to parse macomber ids in format: MAC0001-A
+    macomber_id_re = re.compile(r'^MAC(\d{4})(-[A-F][1-2]?)?')
+
+    # regex to parse manuscript id and folio start/end
+    # Typically looks like:
+    #   41.6 (18v-19v) or 2233(26a)
+    # Also handles rv notation:
+    #   46.62 (65rv)
+    # Will match on pages with "bis", but currently ignores the bis
+    #   2059(20b bis)
+    folio_regex = r'(?P<start>\d+[rvab]) ?[-–]? ?(?P<end>(\d+)?[rvab])?( bis)?'
+    folio_re = re.compile(folio_regex)
+    mss_id_re = re.compile(r'^(?P<id>[\w.]+) ?\(' + folio_regex + r'\)')
+
+    # translate collection names from the macomber file
+    # to the form needed for the spreadsheet
+    collection_lookup = {
+        'EMDL': 'EMDL (HMML)',
+        'EMIP': 'EMIP (HMML)',
+        'AECE': 'AECE (HMML)',
+        'CBS': 'C-Berlin (BS)',
+        'CCBE': 'C-Dublin (CBL)',
+        'CF': 'C-Florence (BNCF)',
+        'CL': 'C-Leningrad (RAN)',
+        'CRA': 'CR-Paris (BNF)',
+        'DULE': 'Duke (Duke)',
+        'EMML': 'EMML (HMML)',
+        'G': 'C-Veroli (BGV)',
+        'GBAE': 'G-Milan (BAM)',
+        'GVE': 'G-Vatican (BAV)',
+        'HBS': 'H-Berlin (BS)',
+        'LUE': 'L-Uppsala (UU)',
+        'SALE': 'S-Rome (ANL)',
+        'SBLE': 'S-London (SBL)',
+        'SGE': 'S-Paris (SBN)',
+        'SWE': 'M-SWE (SWE)',
+        'VLVE': 'VL-Vatican (BAV)',
+        'WBLE': 'W-London (BM)',
+        'PEth': 'PEth (PEth)',    # should this be PUL ?
```

Comment:
makese sense to me for it to be PUL. I feel like we've talked about this before and agreed that it would be PUL?

Reply:
I think so, too - this is one of my open questions for Wendy.

Reply:
Wendy [confirmed via slack](https://cdhprinceton.slack.com/archives/CJ2V8M0B0/p1580227112020500) this should be PUL:
> Re PEth, yes, excellent catch! I should be PUL.